### DISCUSSION TITLE
Domain names lost with CombinerGenerator

### DISCUSSION THREAD

**Discussion by adigc** [idaholab/moose#24577] [Q&A Meshing] [answered]
Hello everyone, 

I have a similar problem with `CombinerGenerator` as described here: #20823
Since the discussion is a year old, I am making a new one.

I am trying to combine two meshes. 

- Mesh 1 : Read in from a `.e` file (block name : neck)
- Mesh 2 : Generated from a GeneratedMeshGenerator (block id: 50)

Now when I am trying to combine these two using `CombinerGenerator`, the block names and side set names of the second mesh in the `inputs` are lost. 

**Case 1:** neck, line

```
  [cmbn]
      type = CombinerGenerator
      inputs = 'neck line'
  [] 
```

Output when running with --mesh-only option
![image](https://github.com/idaholab/moose/assets/135338370/73060f04-94ca-4099-83bb-5344612c991d)

Since neck comes before the line, we can see from the output, the block name `neck` is preserved. 

**Case 2:** line, neck

```
  [cmbn]
      type = CombinerGenerator
      inputs = 'line neck'
  []
```

Output when running with --mesh-only option
![image](https://github.com/idaholab/moose/assets/135338370/60a7f1ed-ef0e-4b70-b3de-9c2b66880791)

Since line comes before neck, we can see from the output, the block name `neck` is lost.  Similarly, node set names are also lost. 

I am attaching the above mentioned example here :  

[combiner_mesh_example.zip](https://github.com/idaholab/moose/files/11635411/combiner_mesh_example.zip)

Any pointers on how to solve this will be helpful. 

Thank you in advance.

**Comment by GiudGiud** [comment]
Hello

Are you trying to stitch the meshes?

Or just have them disconnected in the same simulation?

Guillaume

**Reply by adigc** [reply]
Hi @GiudGiud 

for now disconnected, but in my upcoming investigations, I will have to connect their nodes with MPCs probably.

**Reply by adigc** [reply]
In the example I provided above, the two mehes overlap. But in a general case they do not.

**Reply by GiudGiud** [reply]
We're not going to be able to patch this over the next 2 weeks because of the upcoming training.

I would recommend you re-create the subdomains you need using other mesh generators, like the ParsedSubdomainGenerator

**Reply by adigc** [reply]
@GiudGiud thank you for the reply. I believe that I have found necessary changes to the code for this issue. Is it ok if I make a PR with these changes ?

**Reply by GiudGiud** [reply]
That’s a good idea. We can take a look in a PR

**Comment by MengnanLi91** [comment, marked as answer]
I think this issue has already been fixed in #22838.  In your input file [line] doesn't have a proper block name, you will need to use another RenameBlockGenerator to provide a name for "line". Then the both subdomains should have names in the end.
